### PR TITLE
refactor: output ordering

### LINKED ISSUE #8297: Change output ordering display of sources
### Is your feature request related to a problem or challenge?

Current source executors(`CsvExec, etc.`) contains `outpout_ordering` inside display. For the following table
```sql
CREATE EXTERNAL TABLE multiple_ordered_table (
  a0 INTEGER,
  a INTEGER,
  b INTEGER,
  c INTEGER,
  d INTEGER
)
STORED AS CSV
WITH HEADER ROW
WITH ORDER (a ASC, b ASC)
WITH ORDER (c ASC)
LOCATION '../core/tests/data/window_2.csv';
```
display would be 

```sql
----CsvExec: file_groups={1 group: [[path]]}, projection=[a, b], output_ordering=[a@0 ASC NULLS LAST, b@1 ASC NULLS LAST], has_header=true
```

It would be nice, if output ordering would contain all of the valid orderings at the source such as following

```sql
----CsvExec: file_groups={1 group: [[path]]}, projection=[a, b], output_orderings=[[a@0 ASC NULLS LAST, b@1 ASC NULLS LAST], [c@2 ASC NULLS LAST]], has_header=true
```

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by QuenKar**
## Which issue does this PR close?

Closes #8297

## What changes are included in this PR?

refactor the format of output_orderings

### CODE REVIEW

File: datafusion/core/src/datasource/physical_plan/mod.rs [hunk 1]
```diff
@@ -135,10 +135,23 @@ impl DisplayAs for FileScanConfig {
             write!(f, ", infinite_source=true")?;
         }
 
-        if let Some(ordering) = orderings.first() {
-            if !ordering.is_empty() {
-                write!(f, ", output_ordering={}", OutputOrderingDisplay(ordering))?;
+        if !orderings.is_empty() && !orderings[0].is_empty() {
```
Comment (Weijun-H) [inline datafusion/core/src/datasource/physical_plan/mod.rs:138]:
```suggestion
        if let Some(ordering) = orderings.get(0) {
            if !ordering.is_empty() {
```

File: datafusion/core/src/datasource/physical_plan/mod.rs [hunk 2]
```diff
@@ -135,10 +135,23 @@ impl DisplayAs for FileScanConfig {
...
+            let start = if orderings.len() == 1 {
+                ", output_ordering="
+            } else {
+                ", output_orderings=["
+            };
+            write!(f, "{}", start)?;
+            for (idx, ordering) in orderings.iter().enumerate() {
+                if !ordering.is_empty() {
```
Comment (Weijun-H) [inline datafusion/core/src/datasource/physical_plan/mod.rs:146]:
```suggestion
                for (idx, ordering) in
                    orderings.iter().enumerate().filter(|(_, o)| !o.is_empty())
```

**Review by Weijun-H — COMMENTED** [review]
Thanks @QuenKar 👍

File: datafusion/core/src/datasource/physical_plan/mod.rs (hunk 1 above)
Comment (Weijun-H) [reply datafusion/core/src/datasource/physical_plan/mod.rs:138]: Maybe it's more straightforward.

**Review by Weijun-H — APPROVED** [review]
LGTM! Thanks @QuenKar 👍

**Review by alamb — APPROVED** [review]
Thank you @QuenKar  and @Weijun-H  for the review!

**Pull request merged**